### PR TITLE
[6CORE-075] Add reactive database support for Warehouse Service

### PR DESCRIPTION
**Pull request by VictoriaKh-T**

### CODE REVIEW

File: 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+@Getter
+@Setter
+@NoArgsConstructor
+@Table(name = "inventories")
+public class InventoryEntity {
+  @Id private Long id;
+
+  @Column("actual_quantity")
+  private int actualQuantity;
```
Comment (sokimaaa) [inline 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java]: use wrapper classes pls for entity
```suggestion
  private Integer actualQuantity;
```

File: 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  @Column("available_quantity")
+  private int availableQuantity;
+
+  @Column("warehouse_id")
+  private String warehouseId;
```
Comment (sokimaaa) [inline 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java]: Remember that this is Entity? (ORM)
```suggestion
  private Warehouse warehouse;
```

**Review by sokimaaa — CHANGES_REQUESTED** [review]

File: 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com._6core.platform.warehouseinfra.adapter.driven.persistence.entity;
+
+import java.util.Optional;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.relational.core.mapping.Column;
+import org.springframework.data.relational.core.mapping.Table;
+
+@Data
```
Comment (sokimaaa) [inline 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java]: Dont use @Data for Entity. 
```suggestion
@Setter
@Getter
```

File: 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  private ProductEntity product;
+
+  public Optional<WarehouseEntity> getWarehouse() {
+    return Optional.ofNullable(this.warehouse);
+  }
+
+  public Optional<ProductEntity> getProduct() {
+    return Optional.ofNullable(this.product);
+  }
```
Comment (sokimaaa) [inline 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java]:
```suggestion
```

File: 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+public class InventoryEntity {
+  @Id private String id;
+
+  @Column("actual_quantity")
+  private Integer actualQuantity;
+
+  @Column("available_quantity")
+  private Integer availableQuantity;
+
+  private WarehouseEntity warehouse;
```
Comment (sokimaaa) [inline 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java]: add @Transient for all dependencies. 

Check this guide to find out why:
https://medium.com/pictet-technologies-blog/reactive-programming-with-spring-data-r2dbc-ee9f1c24848b

**Review by sokimaaa — CHANGES_REQUESTED** [review]

**Comment by sokimaaa** [comment]
Pls verify that your object-relational mapping work! 
Otherwise, what is the goal of this task? 

P.s. For example, create R2dbc repository, try to insert/update/get entities (with Join). To check that all is working! 
P.s.s. if you did some checks -> dont push them.

File: 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java (hunk 5 above)
Comment (VictoriaKh-T) [reply 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java]: thank you, I had doubts about it

Comment (sokimaaa) [reply 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java]: pls in the future do research on your own. 

if you dont understand something - feel free to google (medium, baeldung, stackoverflow, ...)
On the web a lot of working examples.

File: 3-platform/warehouse/infra/src/main/resources/application.properties
Comment (sokimaaa) [inline 3-platform/warehouse/infra/src/main/resources/application.properties:1]: dont create properties in infra/app/domain modules.

**Review by sokimaaa — APPROVED** [review]

File: 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java (hunk 5 above)
Comment (VictoriaKh-T) [reply 3-platform/warehouse/infra/src/main/java/com/_6core/platform/warehouseinfra/adapter/driven/persistence/entity/InventoryEntity.java]: I googled and found it, and came to the conclusion that it is not needed. 
this is an example i found.
зображення

**Review by YuraKhilko — APPROVED** [review]

**Pull request merged**